### PR TITLE
Change onboarding flow in web UI

### PR DESCRIPTION
**Pull request by Gargron**
Simplifies onboarding flow from an index view with 4 actions to a linear flow with 2 screens. Profile setup proceeds to the follow page with a new prominent "Done" button that leads to the home feed. The follow page gains a quick search function.

https://github.com/user-attachments/assets/f011af65-d082-45fd-ac83-d763ceb771c7

### CODE REVIEW

**Comment by shleeable** [comment]
Please consider the more involved onboarding I've raised in #32666

File: app/javascript/mastodon/api_types/suggestions.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export type ApiSuggestionSourceJSON =
+  | 'featured'
+  | 'most_followed'
+  | 'most_interactions'
+  | 'similar_to_recently_followed'
+  | 'friends_of_friends';
+
+export interface ApiSuggestionJSON {
+  sources: [ApiSuggestionSourceJSON, ...ApiSuggestionSourceJSON[]];
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/api_types/suggestions.ts:11]: I'm confused about this syntax. What does this mean?

File: app/javascript/mastodon/components/follow_button.tsx [hunk 2]
```diff
@@ -99,7 +99,12 @@ export const FollowButton: React.FC<{
   return (
     <Button
       onClick={handleClick}
-      disabled={relationship?.blocked_by || relationship?.blocking}
+      disabled={
+        relationship?.blocked_by ||
+        relationship?.blocking ||
+        account?.suspended ||
+        !!account?.moved
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/components/follow_button.tsx:106]: Slightly worried about the moved bit, different implementations have different expectations about what a moved account means. Being completely unable to follow a moved account might be an issue.

File: app/javascript/mastodon/components/account.jsx
Comment (ClearlyClaire) [inline app/javascript/mastodon/components/account.jsx:1]: The `follow`, `unfollow` and `cancel_follow_request` keys of `messages` are not used anymore.

**Review by ClearlyClaire — COMMENTED** [review]
This PR has undocumented changes to the list edition UI, which I honestly find a bit confusing.

**Comment by Gargron** [comment]
The `<ColumnSearchHeader>` is changed in accordance with guidance from Sam and Scott to be an always visible search field. The onboarding and the list members screen that both use this component are changed accordingly.

File: app/javascript/mastodon/components/account.jsx
Comment (Gargron) [reply app/javascript/mastodon/components/account.jsx:1]: Removed!

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**